cypress/integration/table/filtering.spec.ts

```diff
@@ -0,0 +1,114 @@
+describe('Filtering Component', () => {
+  beforeEach(() => {
+    cy.intercept('**/records**', (req) => {
+      req.reply({
+        statusCode: 200,
+        fixture: 'records.json',
+      });
+    }).as('getRecords');
+
+    cy.intercept('**/records/count', (req) => {
+      req.reply({ statusCode: 200, fixture: 'recordCount.json' });
+    }).as('getRecordCount');
+
+    // remove second @getRecords once we query channels properly
+    cy.visit('/').wait(['@getRecords', '@getRecords', '@getRecordCount']);
+  });
+
+  it('opens dialogue when you click on filter button and closes when you click close', () => {
+    cy.contains('Filter').click();
+
+    cy.get('[role="dialog"]').contains('Filters').should('be.visible');
+
+    cy.get('[role="dialog"]').contains('Close').click();
+
+    cy.get('[role="dialog"]').should('not.exist');
+  });
+
+  it('lets a user create a filter', () => {
+    cy.contains('Filter').click();
+
+    cy.get('input[role="combobox"]').type('Time{enter}is not null{enter}');
+
+    cy.get('[role="dialog"]')
+      .contains('[role="button"]', 'Time')
+      .should('be.visible');
+    cy.get('[role="dialog"]')
+      .contains('[role="button"]', 'is not null')
+      .should('be.visible');
+
+    cy.contains('Apply').should('not.be.disabled');
+    cy.contains('Apply').click();
+
+    cy.wait('@getRecords').should(({ request }) => {
+      console.log('req', request);
+      expect(request.url).to.contain('conditions=');
+      expect(request.url).to.contain(
+        'conditions=%7B%22metadata.timestamp%22%3A%7B%22%24ne%22%3Anull%7D%7D'
+      );
```

Comment:
Is there some way we can improve this by parsing some JSON or something down into this format without us having to write it ourselves? Saves us figuring out the percentages and stuff

Reply:
Yeah, I can do some URL escape conversion 👍